libsolidity/analysis/DeclarationContainer.cpp

```diff
@@ -49,16 +49,9 @@ Declaration const* DeclarationContainer::conflictingDeclaration(
 		dynamic_cast<MagicVariableDeclaration const*>(&_declaration)
 	)
 	{
-		// check that all other declarations with the same name are functions or a public state variable or events.
-		// And then check that the signatures are different.
+		// check that all other declarations are of the same kind
 		for (Declaration const* declaration: declarations)
 		{
-			if (auto variableDeclaration = dynamic_cast<VariableDeclaration const*>(declaration))
-			{
-				if (variableDeclaration->isStateVariable() && !variableDeclaration->isConstant() && variableDeclaration->isPublic())
-					continue;
```

Comment:
Judging from the git history this exception was introduced to be able to compile an old version of ``std/StandardToken.sol``. However, this is removed now and the last version in the repository wouldn't even have needed this, so I think it's fine to remove it entirely.